token/js/test/e2e-2022/transferFee.test.ts

```diff
@@ -68,203 +69,281 @@ describe('transferFee', () => {
+            expect(accountInfo.amount).to.eql(TRANSFER_AMOUNT);
+            const transferFeeAmount = getTransferFeeAmount(accountInfo);
+            expect(transferFeeAmount).to.not.be.null;
+            if (transferFeeAmount !== null) {
+                expect(transferFeeAmount.withheldAmount).to.eql(BigInt(0));
+            }
+            const mintInfo = await getMint(connection, mint, undefined, TEST_PROGRAM_ID);
+            const transferFeeConfig = getTransferFeeConfig(mintInfo);
+            expect(transferFeeConfig).to.not.be.null;
+            if (transferFeeConfig !== null) {
+                expect(transferFeeConfig.withheldAmount).to.eql(BigInt(0));
+            }
+        });
+        it('transferFeeConfigAuthority', async () => {
+            await setAuthority(
+                connection,
+                payer,
+                mint,
+                transferFeeConfigAuthority,
+                AuthorityType.TransferFeeConfig,
+                null,
+                [],
+                undefined,
+                TEST_PROGRAM_ID
+            );
+            const mintInfo = await getMint(connection, mint, undefined, TEST_PROGRAM_ID);
+            const transferFeeConfig = getTransferFeeConfig(mintInfo);
+            expect(transferFeeConfig).to.not.be.null;
+            if (transferFeeConfig !== null) {
+                expect(transferFeeConfig.transferFeeConfigAuthority).to.eql(PublicKey.default);
+            }
+        });
+        it('withdrawWithheldAuthority', async () => {
+            await setAuthority(
+                connection,
+                payer,
+                mint,
+                withdrawWithheldAuthority,
+                AuthorityType.WithheldWithdraw,
+                null,
+                [],
+                undefined,
+                TEST_PROGRAM_ID
+            );
+            const mintInfo = await getMint(connection, mint, undefined, TEST_PROGRAM_ID);
+            const transferFeeConfig = getTransferFeeConfig(mintInfo);
+            expect(transferFeeConfig).to.not.be.null;
+            if (transferFeeConfig !== null) {
+                expect(transferFeeConfig.withdrawWithheldAuthority).to.eql(PublicKey.default);
+            }
+        });
     });
-    it('withdrawWithheldAuthority', async () => {
-        await setAuthority(
-            connection,
-            payer,
-            mint,
-            withdrawWithheldAuthority,
-            AuthorityType.WithheldWithdraw,
-            null,
-            [],
-            undefined,
-            TEST_PROGRAM_ID
-        );
-        const mintInfo = await getMint(connection, mint, undefined, TEST_PROGRAM_ID);
-        const transferFeeConfig = getTransferFeeConfig(mintInfo);
-        expect(transferFeeConfig).to.not.be.null;
-        if (transferFeeConfig !== null) {
-            expect(transferFeeConfig.withdrawWithheldAuthority).to.eql(PublicKey.default);
-        }
+
+    describe('with null authorities', () => {
```

Comment:
The diff for the tests looks really funky, but I've simply just refactored the setup function to be re-used down here with the provided authorities, and then just added this whole `with null authorities` section.